packages/web-components/src/components/va-search/va-search.tsx

```diff
@@ -15,159 +16,348 @@ import {
+   * Closes listbox when focus is outside of the host element
    */
-  @Prop() inputValue?: string;
+  private handleBlur = () => {
+    if (!this.formattedSuggestions.length) return;
+    this.isListboxOpen = false;
+  };
 
   /**
-   * The aria-label for search input and button. Default is 'Search'.
+   * Fires a submit event
    */
-  @Prop() label: string = 'Search';
+  private handleSubmit = () => {
+    this.el.dispatchEvent(
+      new CustomEvent('submit', {
+        bubbles: true,
+        cancelable: true,
+        detail: { value: this.inputRef.value },
+      }),
+    );
+  };
 
-  // Input Event Handlers
-  private handleInputBlurEvent = (event: FocusEvent) => {
-    this.inputBlurEvent.emit(event);
+  // Input event handlers
+  /**
+   * Opens listbox when search input has focus and suggestions are provided
+   */
+  private handleInputFocus = () => {
+    if (this.formattedSuggestions.length && !this.isListboxOpen) {
+      this.isListboxOpen = true;
+    }
   };
 
-  private handleInputChangeEvent = (event: Event) => {
-    this.inputChangeEvent.emit(event);
+  /**
+   * Implements keyboard interface from Keyboard Support at
+   * https://www.w3.org/TR/wai-aria-practices-1.1/examples/combobox/aria1.1pattern/listbox-combo.html
+   *
+   * Enter key was added to fire a submit event.
+   * Tab key was added to aid in isListboxOpen state management.
+   */
+  private handleInputKeyDown = (event: KeyboardEvent) => {
+    const options = this.el.shadowRoot.querySelectorAll(
+      '[role="option"]',
+    ) as NodeListOf<HTMLLIElement>;
+    const firstOption = options[0];
+    const lastOption = options[options.length - 1];
+
+    switch (event.key) {
+      case 'ArrowDown':
+        if (!firstOption) return;
+        this.selectSuggestion(firstOption);
+        break;
+      case 'ArrowUp':
+        if (!lastOption) return;
+        this.selectSuggestion(lastOption);
+        break;
+      case 'Enter':
+        this.handleSubmit();
+        break;
+      case 'Escape':
+        this.inputRef.value = '';
+        break;
+      case 'Tab':
+        this.isListboxOpen = false;
+        break;
+      default:
+        break;
+    }
   };
 
-  private handleInputFocusEvent = (event: FocusEvent) => {
-    this.inputFocusEvent.emit(event);
+  // Button event handlers
+  /**
+   * Fires a submit event when search button is clicked
+   */
+  private handleButtonClick = () => {
+    this.handleSubmit();
   };
 
-  private handleInputKeyDownEvent = (event: KeyboardEvent) => {
-    this.inputKeyDownEvent.emit(event);
+  // Listbox event handlers
+  /**
+   * Sets search input value to the suggestion clicked,
+   * removes aria-selected from previously selected suggestion if it exists,
+   * closes the listbox and fires a submit event.
+   */
+  private handleListboxClick = (index: number) => {
+    const suggestion = this.el.shadowRoot.getElementById(
+      `listbox-option-${index}`,
+    );
+    if (!suggestion) return;
```

Comment:
If this accepted the event argument could we use `event.target` for the suggestion instead of using the index to get it?